examples/apps/heart_disease_eval/client/heart_gui.py

```diff
@@ -208,18 +209,22 @@ def evaluate(self, message):
 		# Convert workloadId to hex
 		workload_id = "heart-disease-eval"
 		workload_id = workload_id.encode("UTF-8").hex()
-		wo_submit_json = jrpc_request.WorkOrderSubmitJson(3, 6000, "pformat",
-			worker_id, workload_id, "0x1234", 
+		session_iv = utility.generate_sessioniv()
+		encrypted_session_key = utility.generate_encrypted_session_key(
+			session_iv, worker_obj.encryption_key)
+		requester_nonce = secrets.token_hex(32)
```

Comment:
Use a 128 bit nonce. 32 bits is too short. A nonce is never converted to an integer and is needed to make hashes unique.

Reply:
secrets.token_hex(32) will create random hex string of length 32., this will equal to 16byte or 128bits.

Reply:
You could also use crypto.random_bit_string(NO_OF_BYTES).

Reply:
lets keep secretes because it generates hex string and easy to use.
If i replace random_bit_string() i am getting some other issues, File "/project/TrustedComputeFramework/tools/build/_dev/lib/python3.6/site-packages/tcf_examples_common-0.0.1.dev1-py3.6-linux-x86_64.egg/utility/signature.py", line 134, in __calculate_hash_on_concatenated_string
[11:42:29 WARNING STDERR] workorder_id = (input_json_params['workOrderId']).encode('UTF-8')